substrate/client/network-gossip/src/bridge.rs

```diff
@@ -81,6 +85,7 @@ impl<B: BlockT> GossipEngine<B> {
 	pub fn new<N, S>(
 		network: N,
 		sync: S,
+		notification_service: Box<dyn NotificationService>,
```

Comment:
dq: Would it be possible to use a similar pattern across `NotificationService`, `Network` and `Syncing`?

Reply:
In the SDK refactoring guidelines, it's stated that dynamic dispatch should be preferred over generics. `network` and `sync` can eventually be converted to use dynamic dispatch but I would rather not make `notification_service` into a generic that has to be later converted back to use dynamic dispatch.